src/classes/PSSModel.ts

```diff
@@ -121,46 +122,54 @@ export default class PSSModel {
         return true;
       case "recurring":
         for (const task of this.tasks) {
+          // Ignore Anti-tasks
+          if (task instanceof AntiTask) {
+            continue;
+          }
           if (
-            Frequency.Daily ||
-            (frequency === Frequency.Weekly && dayOfTheWeek(startDate) === dayOfTheWeek(endDate ?? -1)) ||
-            (frequency === Frequency.Monthly && getDayOfMonth(task.startDate) === getDayOfMonth(startDate))
+            (task instanceof RecurringTask || task instanceof TransientTask) &&
+            (
+              (task instanceof TransientTask && task.startDate === startDate && task.startTime === startTime) ||
+              (
+                task instanceof RecurringTask &&
+                (
+                  (frequency === Frequency.Daily) ||
+                  (
+                    frequency === Frequency.Weekly &&
+                    dayOfTheWeek(startDate) === dayOfTheWeek(task.startDate)
+                  ) ||
+                  (
+                    frequency === Frequency.Monthly &&
+                    getDayOfMonth(startDate) === getDayOfMonth(task.startDate)
+                  )
+                )
+              )
+            ) &&
+            (
+              startTime < task.startTime + task.duration &&
+              task.startTime < startTime + duration
+            )
           ) {
-            // Check for overlapping time
-            if (startTime < task.startTime + task.duration && task.startDate < startTime + duration) {
-              return `New recurring tasks conflicts with existing ${task.taskType} task "${task.name}" at ${getDateTime(
-                task.startDate,
-                task.startTime
-              )}`;
-            }
+            return `New recurring task conflicts with existing ${task.taskType} task "${task.name}" at ${getDateTime(
+              task.startDate,
+              task.startTime
+            )}`;
           }
         }
         return true;
       case "anti":
-        // Check for existing antitask (we don't need to check duration)
+        // Check for existing anti-task (we don't need to check duration)
         for (const antiTask of this.tasks.filter((task): task is AntiTask => task instanceof AntiTask)) {
-          if (startDate === antiTask.startDate && startTime === antiTask.startTime && duration === antiTask.duration) {
+          if (startDate === antiTask.startDate && startTime === antiTask.startTime) {
             return `New antitask conflicts with an existing anti-task ${antiTask.name}.`;
           }
         }
-
-        // Check for valid recurring task
```

Comment:
Did this get moved somewhere else? otherwise wouldn't removing this allow the user to create anti-tasks that don't correspond to a valid recurring task?